### ISSUE TITLE
https  + HttpListener doesn't work with MONO_TLS_PROVIDER=btls

### ISSUE THREAD

**Issue by alvarengasv** [mono/mono#16189]
### Steps to Reproduce

1. Create a certificate and register it with httpcfg.
```
makecert -r -eku 1.3.6.1.5.5.7.3.1 -n "CN=MONOWEB" -sv monoweb-8181.pvk monoweb-8181.cer
httpcfg -add -port 8181 -cert monoweb-8181.cer -pvk monoweb-8181.pvk
```
2. Execute the attached solution
`mono monoweb.exe`

3. View http://x.x.x.x:8080 and https://x.x.x.x:8181 with any browser.

### Current Behavior
http works.
https fails with these errors:
Edge: Can’t connect securely to this page
Chrome: ERR_CONNECTION_RESET
Firefox: PR_END_OF_FILE_ERROR

### Expected Behavior
https should show the self-signed certificate warning and allow me to continue.

## When starting with MONO_TLS_PROVIDER=legacy
`MONO_TLS_PROVIDER=legacy mono monoweb.exe`

### Current Behavior
Edge: Works after accepting the security warning.
https fails with these errors:
Chrome: ERR_SSL_PROTOCOL_ERROR
Firefox: SSL_ERROR_RX_MALFORMED_SERVER_HELLO

## On which platforms did you notice this

[x ] Linux

**Version Used**:

Mono JIT compiler version 
[monobug.zip](https://github.com/mono/mono/files/3494765/monobug.zip)
6.0.0.319 (tarball Fri Aug  9 16:23:57 UTC 2019)
[monobug.zip](https://github.com/mono/mono/files/3494767/monobug.zip)

**Comment by baulig** [comment]
Unfortunately, the error handling with HttpListener is not very good.  If anything goes wrong with that certificate, it will just silently ignore it.

**Comment by baulig** [comment]
Should have been fixed with #16212.
